src/caduceus/outboundSender.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"hash"
+	"net/http"
+	"net/url"
+	"regexp"
+	"sync"
+	"time"
+)
+
+type OutboundRequest struct {
+	req         CaduceusRequest
+	event       string
+	transId     string
+	deviceId    string
+	contentType string
+}
+
+type OutboundSenderFactory struct {
+	Url         string
+	ContentType string
+	Client      *http.Client
+	Secret      string
+	Until       int64
+	Events      []string
+	Matchers    map[string][]string
+	NumWorkers  int
+	QueueSize   int
+}
+
+type OutboundSender struct {
+	url          string
+	contentType  string
+	deliverUntil int64
+	dropUntil    int64
+	client       *http.Client
+	secret       []byte
+	events       []*regexp.Regexp
+	matcher      map[string][]*regexp.Regexp
+	queueSize    int
+	queue        chan OutboundRequest
+	wg           sync.WaitGroup
+}
+
+func (osf OutboundSenderFactory) New() (obs *OutboundSender, err error) {
+	if _, err = url.ParseRequestURI(osf.Url); nil != err {
+		return
+	}
+
+	if nil == osf.Client {
+		err = errors.New("nil http.Client")
+		return
+	}
+
+	tmp := &OutboundSender{
+		url:          osf.Url,
+		contentType:  osf.ContentType,
+		client:       osf.Client,
+		deliverUntil: osf.Until,
+		queueSize:    osf.QueueSize,
+	}
+
+	if "" != osf.Secret {
```

Comment:
`if osf.Secret != "" {` is only 1 character away from `if osf.Secret = "" {`  Backwards will let the compiler throw an error, but is a bit different to look at :).